src/content/developers/docs/oracles/index.md

```diff
@@ -8,6 +8,211 @@ incomplete: true
 
 Oracles are a data feed that connect Ethereum to off-chain, real-world information, so you can query data in your smart contracts. For example, prediction market dapps use oracles to settle payments based on events. A prediction market may ask you to bet your ETH on the next president of the United States. They'll use an oracle to confirm the outcome and pay out to the winners.
 
+## Prerequisites {#prerequisites}
+
+Make sure you're familiar with [nodes](/developers/docs/nodes-and-clients/), [consensus mechanisms](/developers/docs/consensus-mechanisms/), and [smart contract anatomy](/developer/docs/smart-contracts/anatomy/), specifically events.
+
+## What is an oracle {#what-is-an-oracle}
+
+Oracles act as a bridge between the blockchain and the real world. They allow you to query APIs to get information into your smart contracts. This could be anything from price information to weather reports.
+
+## Why are they needed? {#why-are-they-needed}
+
+With a blockchain like Ethereum you need every node in the network to be able to replay every transaction and end up with the same result, guaranteed. APIs introduce potentially variable data. If you were sending someone an amount of ETH based on an agreed \$USD value using a price API, the query would return a different result from one day to the next. Not to mention, the API could be hacked or deprecated. If this happens, the nodes in the network wouldn't be able to agree on Ethereum's current state, effectively breaking [consensus](/developers/docs/consensus-mechanisms/).
+
+Oracles solve this problem by posting the data on the blockchain. So any node replaying the transaction will use the same immutable data that's posted for all to see. To do this, an oracle is made up of a smart contract and some off-chain components that can query APIs.
+
+## Oracle architecture {#oracle-architecture}
+
+To understand how an oracle works, let's play through a scenario where your smart contract needs to know who won the superbowl. This is an example of how it could work:
+
+1. Your smart contract requests information from an oracle smart contract.
+2. The oracle smart contract emits an [event](/developers/docs/smart-contracts/anatomy/#events-and-logs).
+3. Off-chain oracle nodes listen for events and upon hearing one, they query an API.
+4. The API returns a JSON response to the nodes.
+5. The nodes call on the oracle smart contract.
+6. The oracle smart contract returns the data to your smart contract.
```

Comment:
This is all within a single Ethereum tx? I don't think that's actually how it works. Let's discuss tomorrow 😄